python/src/trezorlib/cli/trezorctl.py

```diff
@@ -158,20 +163,32 @@ def configure_logging(verbose: int):
 )
 @click.version_option()
 @click.pass_context
-def cli(ctx, path, verbose, is_json, passphrase_on_host, session_id):
+def cli(
+    ctx: click.Context,
+    path: str,
+    verbose: int,
+    is_json: bool,
+    passphrase_on_host: bool,
+    session_id: Optional[str],
+) -> None:
     configure_logging(verbose)
 
+    bytes_session_id: Optional[bytes] = None
     if session_id is not None:
         try:
-            session_id = bytes.fromhex(session_id)
+            bytes_session_id = bytes.fromhex(session_id)
         except ValueError:
             raise click.ClickException(f"Not a valid session id: {session_id}")
 
-    ctx.obj = TrezorConnection(path, session_id, passphrase_on_host)
+    ctx.obj = TrezorConnection(path, bytes_session_id, passphrase_on_host)
+
+
+if TYPE_CHECKING:
+    cli = cast(TrezorctlGroup, cli)  # type: ignore [cli needs to be recasted]
```

Comment:
would this work?
```
def cli_main(...):
    ...

cli = cast(...)
```
this way we're not reassigning the same name, and at runtime `cli is cli_main` so all the code below should still work without modification

Reply:
Works fine, even at runtime. However, by doing this we are mixing together the runtime and type-checking (it requires at least some comment) and runtime relies on the `cast` function from `typing` - most probably nothing to worry about, but still...

Reply:
Done in b70acaf - not sure about the right comment to put there (or if we will even do it like this)

Reply:
> runtime relies on the cast function from typing - most probably nothing to worry about

yes, this is perfectly ok. in full-blown desktop Python typing is part of the stdlib and designed to work fine at runtime -- it's the core codebase where we need to worry

Reply:
in fact in trezorlib we **only** need `if TYPE_CHECKING` for imports which would otherwise be circular. We can declare type aliases at top level, etc.

Reply:
> in fact in trezorlib we only need if TYPE_CHECKING for imports which would otherwise be circular

Good to know. However, it is still good to separate what is used for the application itself and what only for typing purposes - so I would use ` if TYPE_CHECKING` consistently everywhere (so we do not have to think about it and it is a nice pattern)

Reply:
Absolutely true for imports. Not entirely convinced about type aliases, istm it would be fine to keep them on top level.

Reply:
The issue could be when type aliases need to reference the things imported only in `if TYPE_CHECKING` branch, which I think is the case quite often (`trezorlib/tools.py` for example)

Reply:
i'd go with something like, "don't introduce TYPE_CHECKING branch if we only create type aliases, but put in type aliases if we have the branch for imports" ?

Reply:
That could be fine. I am still little afraid of consistency, for example question "should I put the type-alias into `""` when used, so that runtime will not complain about it not being defined, or can I leave it without it, because it is defined on the top level"? With `TYPE_CHECKING` everywhere, we just put it between `""` in all cases and do not have to think about it

Reply:
you can also always put type aliases in quotes :)

Reply:
this is totally immaterial btw. choose whichever style you like, any breakage will be immediately detected by the typechecker